optaplanner-core/src/main/java/org/optaplanner/core/impl/score/director/drools/testgen/TestGenDroolsScoreDirector.java

```diff
@@ -93,6 +97,37 @@ public Score calculateScore() {
     }
 
     @Override
+    public void assertShadowVariablesAreNotStale(Score expectedWorkingScore, Object completedAction) {
+        try {
+            journal.enterAssertMode();
+            super.assertShadowVariablesAreNotStale(expectedWorkingScore, completedAction);
+            journal.exitAssertMode();
+        } catch (IllegalStateException e) {
+            // catch corrupted VariableListener exception and create a minimal reproducing test
+            if (e.getMessage().startsWith("Impossible")) {
+                TestGenCorruptedVariableListenerReproducer reproducer
+                        = new TestGenCorruptedVariableListenerReproducer(e.getMessage(), this);
+                // FIXME this is currently broken. The pruning needs to be smarter and not remove genuine variable
+                // updates that directly affect shadow variables in the last (corrupted) variable listeners update.
+                // If the genuine update is removed the shadow update obviously becomes inconsistent, which leads
+                // to a false positive and the journal no longer reproduces the original issue. This is the current
+                // state.
+                TestGenKieSessionJournal minJournal = TestGenerator.minimize(journal, reproducer);
```

Comment:
Please Jirka, create a JIRA for this if it doesn't exist, so we can plan fixing this sometimes in the future.